client/src/components/relatedItemsAndComparison/ProductCard.jsx

```diff
@@ -1,77 +1,110 @@
+        const defaultStyles = product.data.results.filter((style) => style['default?']);
+        const newDefaultStyle = defaultStyles.length > 0
+          ? defaultStyles[0] : product.data.results[0];
+        setDefaultStyle(newDefaultStyle);
+
+        if (newDefaultStyle.photos[0].thumbnail_url !== null) {
+          setImages(newDefaultStyle.photos);
         } else {
-          setDefaultStyle(product.data.results[0]);
+          setImages([{
+            thumbnail_url: 'https://www.freeiconspng.com/uploads/no-image-icon-6.png',
+            url: 'https://www.freeiconspng.com/uploads/no-image-icon-6.png',
+          }]);
         }
       })
       .catch((err) => console.error(err));
-    }
-  }
+  };
+
+  // get info for the current related product (needed for )
+  const getRelatedProduct = () => {
+    axios.get(`/products/${productId}`)
+      .then((product) => {
+        setRelatedProduct(product.data);
+        // getRating();
+      })
+      .catch((err) => console.error(err))
+      .then(() => getStyles());
+  };
 
+  useEffect(() => {
+    if (productId) {
+      getRelatedProduct();
+    }
+    // getStyles();
+  }, [productId]);
 
   useEffect(() => {
-    getRelatedProduct();
-    // getDefaultRelatedProduct();
-  }, []);
+    if (images.length > 0) {
+      setMainImg(images[0].thumbnail_url);
+    }
+  }, [images]);
+
+  // const handleNavigate = () => {
+  //   setProduct(productId);
+  // };
 
   return (
-    <div>
-      {
-        !defaultStyle.photos[0].thumbnail_url
-        ?
-        <img src='https://www.freeiconspng.com/uploads/no-image-icon-6.png'/>
-        :
-        <img src={`${defaultStyle.photos[0].thumbnail_url}`}/>
-      }
-      <h4>{relatedProduct.name}</h4>
+    // add onClick={handleNavigate} to the first div
+    <div
+      className={index === idx ? 'active product-card' : 'product-card'}
+      ref={index === idx ? ref : null}
+    >
+      <img className="mainImg" src={mainImg} alt="Missing" />
+      <ThumbnailsList images={images} setMainImg={setMainImg} />
+      {/* add onClick={handleCompare} to the button */}
+      <button className="compare">Compare</button>
       <div className="category">{relatedProduct.category}</div>
-      <div className="defaultPrice">{relatedProduct.default_price}</div>
-      {/*
-      {starRating} */}
+      <h4 className="product-name">{relatedProduct.name}</h4>
+      {defaultStyle.sale_price
+        ? (
+          <div className="price">
+            <span className="sale">
+              {`$${defaultStyle.sale_price}`}
+            </span>
+            <span className="struckthrough">
+              {`$${defaultStyle.original_price}`}
+            </span>
+          </div>
+        )
+        : (
+          <div className="price default">
+            {`$${defaultStyle.original_price}`}
+          </div>
+        )}
+      <span className="rating">
+        {/* <StarRating rating={rating[0]} count={rating[1]} link={}/> */}
```

Comment:
I haven't implemented anything with the link, you should be fine to leave it out of the props.